### PR TITLE
feat(test): added cypress tests

### PR DESCRIPTION
**Pull request by serge1peshcoff**
I've split my big PR on `new-core-refactor` branch to leave Cypress out of it, and moved it here.

This PR shouldn't change the frontend behaviour, it only should add Cypress tests and bootstrapping. (also it probably won't work as tests are tied to the new changes, but whatever).

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+// Cypress.Commands.add("dismiss", { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
+
+import moment from 'moment'
+
+Cypress.Commands.add('resetDatabase', () => {
+  cy.exec('(cd .. && ./helper.sh --docker -- exec -T oms-core-js sh -c \\\'npm run db:clear \\&\\& npm run db:seed\\\')')
```
Comment (WikiRik) [inline cypress/support/commands.js:30]: Is it possible to run Cypress tests on prod?

File: package.json [hunk 2]
```diff
@@ -12,7 +12,12 @@
   "scripts": {
     "serve": "vue-cli-service serve",
     "build": "vue-cli-service build",
-    "lint": "vue-cli-service lint --no-fix"
+    "build-and-test": "NODE_ENV=test npm run build && npm test",
+    "lint": "vue-cli-service lint --no-fix .",
```
Comment (WikiRik) [inline package.json:16]: Why is the lint check changed to include a period at the end?

File: package.json [hunk 3]
```diff
@@ -12,7 +12,12 @@
...
+    "test": "npm run cypress:run && npm run nyc:report",
+    "cypress:run": "cypress run",
+    "cypress:open": "cypress open",
+    "nyc:report": "nyc report"
```
Comment (WikiRik) [inline package.json:20]: Is this report part of Cypress?

File: cypress/support/commands.js (hunk 1 above)
Comment (serge1peshcoff) [reply cypress/support/commands.js:30]: what do you mean?

File: package.json (hunk 2 above)
Comment (serge1peshcoff) [reply package.json:16]: the last argument should be a folder to check, `.` means current folder here

File: cypress/support/commands.js (hunk 1 above)
Comment (WikiRik) [reply cypress/support/commands.js:30]: Is there a possibility that the Cypress tests will reset the database on prod?

File: package.json (hunk 3 above)
Comment (serge1peshcoff) [reply package.json:20]: here's how it's working:
- `npm run cypress:run` runs tests and collects code coverage
- `npm run nyc:report` generates a code coverage report based on the test runs

I've put it as a part of `npm test` so we'll get code coverage after each run.

File: cypress/support/commands.js (hunk 1 above)
Comment (serge1peshcoff) [reply cypress/support/commands.js:30]: yes, but we never run tests on prod :D this is for CI and local development only

**Pull request merged**